serial_protocol.md

```diff
@@ -0,0 +1,61 @@
+#Protocol:
+During the first use, or when the chiplet changes environments a “NETWORK” call is expected to initialize the wifi parameters.
+
+Every time a serial connection is established we will expect a “INIT” call after which any subsequent calls can be made, until the connection is closed. 
+
+In the following examples we use $ to represent variables. For example $Host represents your firebase host.
+##NETWORK
+Only needs to be called when the chiplet is in a new environment and needs to connect to a new network. This setting will be stored by the chiplet and persisted through power cycles.
+###Usage
+	NETWORK $SSID
+	NETWORK $SSID $PASSWORD
+###Response
+	OK - When connected to new network
+	FAIL - When unable to connect
+###Examples
+	>> NETWORK home-guest
+	<< OK
+	>> NETWORK home-private MySecretPassword
+	<< OK
+
+##INIT
+Must be called after creating a Serial connection, it can take either just a host for accessing public variables or you may also provide a secret for accessing protected variables in the database.
+###Usage
+	INIT $Host
+	INIT $Host $Secret
+###Response
+	OK - Accepted initialization parameters
+###Examples
+	>> INIT https://samplechat.firebaseio-demo.com
+	<< OK
+	>> INIT https://samplechat.firebaseio-demo.com nnz...sdf
+	<< OK
+##GET
+Fetches the value, in json, at $Path and returns it on the serial line.
+###Usage
+	GET $PATH
+###Response
+	$DATA_BYTE_COUNT $DATA
```

Comment:
> They use the first byte of the response to dictate a type, being error, string, blob(bulkstring). 

Nice! We used something similar for error and logging in the [old serial bridge] https://github.com/googlesamples/firebase-arduino/commit/566c3031083f05b59f3f86bc1de79f247c941bd4#diff-dc3ebc3d309c06217829e6e5ed607be0R71, everything that started with a `!` could be ignored.

I'm wondering if we need to "always" prefix the type since the chiplet is meant to be used directly with Serial.[readStringUntil](https://www.arduino.cc/en/Reference/StreamReadStringUntil) and it's like that you already know if you want to read a json number, a string or a json body depending on the path you're `GET`ing.

It would look something like:
```
Serial.println("GET /foo");
float foo = atof(Serial.readStringUntil("\n").c_str());
```